File: basics/25-search-linked-list/Solution.py
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

def search(head: ListNode, target: int) -> bool:
    while head:
        if head.val == target:
            return True
        head = head.next
    return False

File: basics/25-search-linked-list/Solution.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) { this.val = val; }
}

class Solution {
    public boolean search(ListNode head, int target) {
        while (head != null) {
            if (head.val == target) return true;
            head = head.next;
        }
        return false;
    }
}
